cli/azd/pkg/output/formatter.go

```diff
@@ -36,3 +39,36 @@ func NewFormatter(format string) (Formatter, error) {
 		return nil, fmt.Errorf("unsupported format %v", format)
 	}
 }
+
+type contextKey string
+
+const (
+	formatterContextKey contextKey = "formatter"
+	writerContextKey    contextKey = "writer"
+)
+
+func WithFormatter(ctx context.Context, formatter Formatter) context.Context {
+	return context.WithValue(ctx, formatterContextKey, formatter)
+}
+
+func GetFormatter(ctx context.Context) Formatter {
+	formatter, ok := ctx.Value(formatterContextKey).(Formatter)
+	if !ok {
+		return &NoneFormatter{}
```

Comment:
(nit) This illustrates the pitfall of storing data in "context-like" struct and having getters like this one. The question is: is it *always* the case that the "none" formatter is the right default? If not, this is masking bugs where formatter should have been set on the context, but it wasn't.

Reply:
The way I think about it is that if you ask for the formatter and it hasn't been set, then the `None` formatter is the correct choice.  Unless we want to create the concept of a `NotSpecified` formatter or just check against a `nil` value before its use?

Reply:
Up until now we had a concept of "default" format for a command e.g. https://github.com/Azure/azure-dev/blob/main/cli/azd/cmd/deploy.go#L43. So if the format is not set, the default is assumed, which is not always "none". E.g. for `env get-values` command it is "env vars" fromat: https://github.com/Azure/azure-dev/blob/main/cli/azd/cmd/env.go#L49

Reply:
As part of the command builder it does respect the default formatter for the given command.  It calls `output.GetCommandFormatter(cmd)` to get any formatter set by the user or the default formatter if not specified.

Only the cases where a formatter is not specified will it ever default to this `None` formatter now.

I've made a couple tweaks to only set the formatter if found from the command.  The `GetFormatter()` already returns the `None` formatter if not found.

Lets try this out for now and see if it causes us any issues.